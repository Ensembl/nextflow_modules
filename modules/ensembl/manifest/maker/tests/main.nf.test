nextflow_process {

    name "Test Process MANIFEST_MAKER"
    script "modules/ensembl/manifest/integrity/main.nf"

    process "MANIFEST_MAKER"

    tag "modules"
    tag "modules_nfcore"
    tag "manifest"
    tag "manifest/maker"

        when {
            options "-stub"

            process {
                """
                input[0] = Channel.of(
                        '##gff-version 3',
                        'chr1\tID=gene1'
                    )
                    .collectFile(name: 'genes.gff3', newLine: true)
                    .map { file -> [ [ id:'test' ], file ] }
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
                //TODO nf-core: Add all required assertions to verify the test output.
                // See https://nf-co.re/docs/contributing/tutorials/nf-test_assertions for more information and examples.
            )
        }

    }

    // TODO nf-core: Change the test name preferably indicating the test-data and file-format used but keep the " - stub" suffix.
    test("sarscov2 - bam - stub") {

        options "-stub"

        when {
            process {
                """
                // TODO nf-core: define inputs of the process here. Example:
                
                input[0] = [
                    [ id:'test', single_end:false ], // meta map
                    file(params.modules_testdata_base_path + 'genomics/sarscov2/illumina/bam/test.paired_end.sorted.bam', checkIfExists: true),
                ]
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }
}
