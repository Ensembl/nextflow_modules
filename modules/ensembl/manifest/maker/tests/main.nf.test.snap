{
    "sarscov2 - bam - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "manifest.json:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    "versions.yml:md5,d179a9314cfcc5bec5a00d0335d2760b"
                ],
                "versions": [
                    "versions.yml:md5,d179a9314cfcc5bec5a00d0335d2760b"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.3"
        },
        "timestamp": "2025-06-17T11:28:41.159067568"
    },
    "Maker check stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        [
                            "genes.gff3:md5,5e2fd50352e7d2ebec2bdccaa352e541",
                            "manifest.json:md5,d41d8cd98f00b204e9800998ecf8427e"
                        ]
                    ]
                ],
                "1": [
                    "versions.yml:md5,d179a9314cfcc5bec5a00d0335d2760b"
                ],
                "versions": [
                    "versions.yml:md5,d179a9314cfcc5bec5a00d0335d2760b"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.3"
        },
        "timestamp": "2025-06-17T11:28:27.183348913"
    }
}